src/SdkCommon/ClientRuntime/ClientRuntime/RetryAfterDelegatingHandler.cs

```diff
@@ -116,5 +117,37 @@ public RetryAfterDelegatingHandler(HttpMessageHandler innerHandler)
                 return response ?? previousResponseMessage;
             } while (true);
         }
+
+
+        private double GetRetryAfterSeconds(string retryAfterValue)
+        {
+            // retryAfterValue can have three formats: seconds, timespan or a date.
+            bool isSeconds = double.TryParse(retryAfterValue, out double retrySeconds);
+            if (!isSeconds)
+            {
+                bool isTimeSpan = TimeSpan.TryParse(retryAfterValue, out TimeSpan span);
+
+                if (isTimeSpan)
+                {
+                    retrySeconds = span.TotalSeconds;
+                }
+                else
+                {
+                    bool isDate = DateTime.TryParse(retryAfterValue, out DateTime date);
+
+                    if (isDate)
+                    {
+                        retrySeconds = date.Subtract(DateTime.Now).TotalSeconds;
+                    }
+                    else
+                    {
+                        // Retry-After header could not be parsed. Defaulting to 60 seconds.
+                        retrySeconds = 60;
```

Comment:
Looks like desired retrySeconds [is 20 seconds](https://github.com/Azure/autorest/issues/2601).